lib/metadata/models/object/object.js

```diff
@@ -0,0 +1,465 @@
+    }
+
+    /**
+     * Returns server side encryption
+     *
+     * @return {string} server side encryption
+     */
+    getAmzServerSideEncryption() {
+        return this.data['x-amz-server-side-encryption'];
+    }
+
+    /**
+     * Set encryption key id
+     *
+     * @param {string} keyId - Encryption key id
+     * @return {ObjectMD} itself
+     */
+    setAmzEncryptionKeyId(keyId) {
+        this.data['x-amz-server-side-encryption-aws-kms-key-id'] = keyId;
+        return this;
+    }
+
+    /**
+     * Returns encryption key id
+     *
+     * @return {string} Encryption key id
+     */
+    getAmzEncryptionKeyId() {
+        return this.data['x-amz-server-side-encryption-aws-kms-key-id'];
+    }
+
+    /**
+     * Set encryption customer algorithm
+     *
+     * @param {string} algo - Encryption customer algorithm
+     * @return {ObjectMD} itself
+     */
+    setAmzEncryptionCustomerAlgorithm(algo) {
+        this.data['x-amz-server-side-encryption-customer-algorithm'] = algo;
+        return this;
+    }
+
+    /**
+     * Returns Encryption customer algorithm
+     *
+     * @return {string} Encryption customer algorithm
+     */
+    getAmzEncryptionCustomerAlgorithm() {
+        return this.data['x-amz-server-side-encryption-customer-algorithm'];
+    }
+
+    /**
+     * Set access control list
+     *
+     * @param {object} acl - Access control list
+     * @param {string} acl.Canned -
+     * @param {string[]} acl.FULL_CONTROL -
+     * @param {string[]} acl.WRITE_ACP -
+     * @param {string[]} acl.READ -
+     * @param {string[]} acl.READ_ACP -
+     * @return {ObjectMD} itself
+     */
+    setAcl(acl) {
+        this.data.acl = acl;
+        return this;
+    }
+
+    /**
+     * Returns access control list
+     *
+     * @return {object} Access control list
+     */
+    getAcl() {
+        return this.data.acl;
+    }
+
+    /**
+     * Set object key
+     *
+     * @param {string} key - Object key
+     * @return {ObjectMD} itself
+     */
+    setKey(key) {
+        this.data.key = key;
+        return this;
+    }
+
+    /**
+     * Returns object key
+     *
+     * @return {string} object key
+     */
+    getKey() {
+        return this.data.key;
+    }
+
+    /**
+     * Set location
+     *
+     * @param {string[]} location - location
+     * @return {ObjectMD} itself
+     */
+    setLocation(location) {
+        this.data.location = location;
+        return this;
+    }
+
+    /**
+     * Returns location
+     *
+     * @return {string[]} location
+     */
+    getLocation() {
+        return this.data.location;
+    }
+
+    /**
+     * Set custom headers
+     *
+     * @param {object} data - Headers
+     * @return {ObjectMD} itself
+     */
+    setHeaders(data) {
+        Object.keys(data).forEach(key => { this.data[key] = data[key]; });
+        return this;
+    }
+
+    /**
+     * Returns metadata object
+     *
+     * @return {object} metadata object
+     */
+    getData() {
```

Comment:
minor: since we have a public method for accessing the property, you could name the 'data' property '_data' to follow convention, similar to what we do in BucketInfo class